std/experimental/ndslice/slice.d

```diff
@@ -1767,6 +1767,184 @@ struct Slice(size_t _N, _Range)
         assert(a != [[9, 2], [3, 4]]);
     }
 
+    /++
+    Computes hash value using MurmurHash3 algorithms without the finalization step.
+    Built-in associative arrays have the finalization step.
+
+    Returns: Hash value type of `size_t`.
+
+    See_also: $(LREF Slice.toMurmurHash3), $(MREF std, _digest, murmurhash).
+    +/
+    size_t toHash() const
+    {
+        static if (size_t.sizeof == 8)
+        {
+            auto ret = toMurmurHash3!128;
+            return ret[0] ^ ret[1];
+        }
+        else
+        {
+            return toMurmurHash3!32;
+        }
+    }
+
+    static if (doUnittest)
+    ///
+    pure nothrow @nogc @safe
+    unittest
+    {
+        import std.experimental.ndslice.selection : iotaSlice;
+        const sl = iotaSlice(3, 7);
+        size_t hash = sl.toHash;
```

Comment:
small FYI - using `assert` on ddoced tests isn't bad - in fact a soon-to-come update might rewrite the `assert`s to writeln in interactive mode (see https://github.com/dlang/dlang.org/pull/1297)